openedx/features/course_experience/views/course_outline.py

```diff
@@ -30,26 +31,32 @@ class CourseOutlineFragmentView(EdxFragmentView):
     Course outline fragment to be shown in the unified course view.
     """
 
-    def render_to_fragment(self, request, course_id=None, page_context=None, **kwargs):
+    def render_to_fragment(self, request, course_id, user_is_enrolled=True, **kwargs):  # pylint: disable=arguments-differ
         """
         Renders the course outline as a fragment.
         """
         course_key = CourseKey.from_string(course_id)
-        course_overview = get_course_overview_with_access(request.user, 'load', course_key, check_if_enrolled=True)
+        course_overview = get_course_overview_with_access(
+            request.user, 'load', course_key, check_if_enrolled=user_is_enrolled
+        )
```

Comment:
Not sure what's going on here -- we check to see if they're enrolled if we already know they're enrolled?

Reply:
Hmm. Maybe this can better expressed. The current code is `check_if_enrolled=True` so this was an attempt to keep that logic unless `COURSE_ENABLE_UNENROLLED_ACCESS_FLAG` was on for this course (in which cause this would only be `False` is an unenrolled user was accessing the outline).

Is it okay to just set it to `False`?